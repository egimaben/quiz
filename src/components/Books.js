import React from 'react';
import PropTypes from 'prop-types';
import Book from './Book'

const Books = ({
    authorData,
    onWrongAnswer,
    onRightAnswer,
    bookStyle,
    selectedBook,
    showContinueButton,
    onContinue }) => {
    return <div className='books'>
      {
        authorData.bookOptions.map(
          (bookOption, index) => <Book
            onWrongAnswer={onWrongAnswer}
            onRightAnswer={onRightAnswer}
            bookStyle={bookStyle}
            selectedBook={selectedBook}
            key={index}
            title={bookOption} authorData={authorData} />
        )
      }
      {
        showContinueButton && <button onClick={onContinue} className='continue-button'>Continue</button>
      }
    </div>
  }
  Books.propTypes = {
    authorData: PropTypes.shape({
      bookOptions: PropTypes.arrayOf(PropTypes.string),
      authoredBook: PropTypes.string.isRequired,
      avatar: PropTypes.string.isRequired,
      authorName: PropTypes.string
    })
  }
  export default Books;