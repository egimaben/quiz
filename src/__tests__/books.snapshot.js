'use strict'
import React from 'react';
import renderer from 'react-test-renderer';
import Books from '../components/Books';
it('displays books component correctly',()=>{
    let  authorData={bookOptions:[],authoredBook:'',avatar:''};
    let onContinue=jest.fn();
    //when should not show continue button
    const tree = renderer.create(
        <Books onContinue={onContinue} showContinueButton={false} authorData={authorData}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
//when should show continue button

    const tree2 = renderer.create(
        <Books onContinue={onContinue} showContinueButton={true} authorData={authorData}/>
    ).toJSON();
    expect(tree2).toMatchSnapshot();
})