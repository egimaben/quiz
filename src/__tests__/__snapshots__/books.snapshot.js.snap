// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays books component correctly 1`] = `
<div
  className="books"
/>
`;

exports[`displays books component correctly 2`] = `
<div
  className="books"
>
  <button
    className="continue-button"
    onClick={[MockFunction]}
  >
    Continue
  </button>
</div>
`;
